napari/layers/image/image.py

```diff
@@ -759,9 +763,12 @@ def _set_view_slice(self):
             self, image_indices, image, thumbnail_source
         )
         self._load_slice(data)
-        if self._keep_auto_contrast or self._should_calc_clims:
+        if self._should_calc_clims:
+            self.reset_contrast_limits_range()
             self.reset_contrast_limits()
             self._should_calc_clims = False
+        if self._keep_auto_contrast:
+            self.reset_contrast_limits()
```

Comment:
Is there any case where we would call `self.reset_contrast_limits()` twice, and try not to do that ?

Reply:
Yes, but it will only happen will happen if:
- this is the first slice ever loaded in this layer
- the layer has auto-contrast enabled

But I guess it's as simple as changing to an `elif` and we can avoid even this niche issue.